test/nerves/cache_test.exs

```diff
@@ -32,4 +32,32 @@ defmodule Nerves.CacheTest do
       assert Cache.valid?(package)
     end)
   end
+
+  test "dl folder is queried prior to calling resolver" do
```

Comment:
Is it easy to test the case where the file exists, but has a bad checksum? That seems useful to verify. I'm on the fence on what happens on a failure, but slightly leaning towards Buildroot's policy of deletion.